File: src/main/java/com/example/demo/mapper/ExampleMapper.java

package com.example.demo.mapper;

import com.example.demo.dto.ExampleDto;
import com.example.demo.entity.ExampleEntity;

public class ExampleMapper {

    public static ExampleEntity toEntity(ExampleDto dto) {
        ExampleEntity entity = new ExampleEntity();
        entity.setName(dto.getName());
        entity.setDescription(dto.getDescription());
        return entity;
    }

    public static ExampleDto toDto(ExampleEntity entity) {
        ExampleDto dto = new ExampleDto();
        dto.setName(entity.getName());
        dto.setDescription(entity.getDescription());
        dto.setId(entity.getId());
        return dto;
    }

    public static void updateEntityFromDto(ExampleEntity entity, ExampleDto dto) {
        if (dto.getName() != null && !dto.getName().isBlank()) {
            entity.setName(dto.getName());
        }
        if (dto.getDescription() != null) {
            entity.setDescription(dto.getDescription());
        }
    }
}

File: src/main/kotlin/com/example/demo/mapper/ExampleMapper.kt
package com.example.demo.mapper

import com.example.demo.dto.ExampleDto
import com.example.demo.entity.ExampleEntity

object ExampleMapper {
    fun toEntity(dto: ExampleDto): ExampleEntity {
        val entity = ExampleEntity()
        dto.name?.let { entity.name = it }
        entity.description = dto.description
        return entity
    }

    fun toDto(entity: ExampleEntity): ExampleDto {
        return ExampleDto(
            id = entity.id,
            name = entity.name,
            description = entity.description
        )
    }

    fun updateEntityFromDto(entity: ExampleEntity, dto: ExampleDto) {
        dto.name?.takeIf { it.isNotBlank() }?.let { entity.name = it }
        dto.description?.let { entity.description = it }
    }
}
